faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -392,17 +373,16 @@ private void handleQueryErrors(FullHttpResponse response) {
 
   private <V> CompletableFuture<V> handleNetworkExceptions(CompletableFuture<V> f) {
       return f.whenComplete((v, ex) -> {
-              if (ex instanceof ConnectException ||
-                         ex instanceof TimeoutException) {
-                  throw new UnavailableException(ex.getMessage(), ex);
-              }
-          });
+          if (ex instanceof ConnectException || ex instanceof TimeoutException) {
+              throw new UnavailableException(ex.getMessage(), ex);
+          }
+      });
   }
 
-  private JsonNode parseResponseBody(FullHttpResponse response) throws IOException {
-    JsonNode body = json.readTree(new ByteBufInputStream(response.content()));
+  private JsonNode parseResponseBody(HttpResponse<String> response) throws JsonProcessingException, IllegalArgumentException {
+    JsonNode body = json.readTree(response.body());
     if (body == null) {
-      throw new IOException("Invalid JSON.");
+      throw new IllegalArgumentException("Invalid JSON.");
```

Comment:
in the long run it's probably better to use a dedicated exception class rather than IAE, but overall this looks much better.